### PR TITLE
Resolve #189 Implement MultipleWhitespaceFinder for Rule S6326

### LINKED ISSUE #189: Implement MultipleWhitespaceFinder for Rule S6326
Related to [SONARPHP-1275](https://jira.sonarsource.com/browse/SONARPHP-1275)

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: regex-parsing/src/test/resources/finders/MultipleWhitespaceFinder.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- 'Hello,   world!'         # Noncompliant {{Replace spaces with quantifier `{2}`.}}
+- 'Hello,  world!'          # Noncompliant {{Replace spaces with quantifier `{1}`.}}
+- 'Hello, world!   '        # Noncompliant
+
+- 'Hello, {3}world!'        # Compliant with quantifier
```
Comment (stanislav-jordanov-sonarsource) [inline regex-parsing/src/test/resources/finders/MultipleWhitespaceFinder.yml:6]: minor: Why not have a compliant test with single white spaces separated by non-white space,
e.g.: "Hello , world"

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/MultipleWhitespaceFinder.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  @Override
+  public void visitSequence(SequenceTree sequence) {
+
+    if (!sequence.activeFlags().contains(Pattern.COMMENTS)) {
+      Deque<RegexTree> whitespaces = new ArrayDeque<>();
+      for (RegexTree item : sequence.getItems()) {
+        if (isWhitespace(item)) {
+          whitespaces.add(item);
+        } else if (whitespaces.size() > 1) {
+          reportMultipleWhitespaces(sequence, whitespaces);
```
Comment (stanislav-jordanov-sonarsource) [inline regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/MultipleWhitespaceFinder.java:54]: I think you should have `whitespaces.clear()` right after line 54.
Don't you?

**Review by stanislav-jordanov-sonarsource — CHANGES_REQUESTED** [review]
There's only one serious (potential) issue,
and two minor:
1. suggested additional 2 test cases (unsure if it makes sens)
2. You have not included the task id in the PR comment (the way you taught me to)

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/MultipleWhitespaceFinder.java (hunk 2 above)
Comment (nils-werner-sonarsource) [reply regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/MultipleWhitespaceFinder.java:54]: I did the clear in the reportMultipleWhitespaces method. But I think doing it right after the call is more readable.

**Review by stanislav-jordanov-sonarsource — APPROVED** [review]

**Pull request merged**